cloudbus.py

```diff
@@ -41,23 +45,37 @@
 apiKey = ""
 
 def get_oauth_token():
-    creds = b64encode(clientId + ':' + apiKey)
+    data = clientId + ':' + apiKey
 
-    url = "http://" + CBUS_IP + GET_TOKEN
-    headers = {'Authorization': 'Basic ' + creds}
-    body = urllib.urlencode({'grant_type': 'client_credentials'})
+    # Handle differently for Python3

```

Comment:
Why do we need to handle this differently here?  Isn't it sufficient to deal with the version difference only in get_response()?

Reply:
The b64encode function is different between Python 2 and 3, (error below) and also python 3 uses urllib.parse.urlencode instead of urllib.urlencode. What I have comitted works for python 2 and 3 but there may be a better way.
C:\Users\jimma\Documents\cloudbus_data_analysis\cloudbus_data_analysis> python .\plot_attribute.py intwine_nyce_motion battery_voltage
Traceback (most recent call last):
  File ".\plot_attribute.py", line 17, in <module>
    myDevice = cbDevice()
  File "C:\Users\jimma\Documents\cloudbus_data_analysis\cloudbus_data_analysis\cloudbus.py", line 117, in __init__
    token = get_oauth_token()
  File "C:\Users\jimma\Documents\cloudbus_data_analysis\cloudbus_data_analysis\cloudbus.py", line 68, in get_oauth_token
    creds = b64encode(data)
  File "C:\Users\jimma\AppData\Local\Programs\Python\Python36\lib\base64.py", line 58, in b64encode
    encoded = binascii.b2a_base64(s, newline=False)
TypeError: a bytes-like object is required, not 'str'

Reply:
wow...thats obnoxious.

Ok, just move url out of the if statement and we'll call it a day